test_positive_delete_manifest: AssertionError: assert_no_error: error: '404 Not Found'

session = <airgun.session.Session object at 0x7f1859cc7438>

    @run_in_one_thread
    @skip_if_not_set('fake_manifest')
    @tier2
    def test_positive_delete_manifest(session):
        """Check if deleting a manifest removes it from Activation key
    
        :id: 512d8e41-b937-451e-a9c6-840457d3d7d4
    
        :Steps:
            1. Create Activation key
            2. Associate a manifest to the Activation Key
            3. Delete the manifest
    
        :expectedresults: Deleting a manifest removes it from the Activation
            key
    
        :CaseLevel: Integration
        """
        # Upload manifest
        org = entities.Organization().create()
        with manifests.clone() as manifest:
            upload_manifest(org.id, manifest.content)
        # Create activation key
        activation_key = entities.ActivationKey(
            organization=org,
        ).create()
        # Associate a manifest to the activation key
        subscription = entities.Subscription(organization=org).search(query={
            'search': 'name="{}"'.format(DEFAULT_SUBSCRIPTION_NAME)})[0]
        activation_key.add_subscriptions(data={
            'quantity': 1,
            'subscription_id': subscription.id,
        })
        with session:
            session.organization.select(org.name)
            # Verify subscription is assigned to activation key
            ak = session.activationkey.read(activation_key.name)
            assert (
                ak['subscriptions']['resources']['assigned'][0]['Repository Name']
                == DEFAULT_SUBSCRIPTION_NAME
            )
            # Delete the manifest
>           session.subscription.delete_manifest()

tests/foreman/ui/test_activationkey.py:1317: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../shiningpanda/jobs/1035602e/virtualenvs/d41d8cd9/lib/python3.6/site-packages/airgun/entities/subscription.py:79: in delete_manifest
    'Delete Manifest', has_manifest=False, ignore_error_messages=ignore_error_messages)
../../shiningpanda/jobs/1035602e/virtualenvs/d41d8cd9/lib/python3.6/site-packages/airgun/entities/subscription.py:39: in _wait_for_process_to_finish
    view.flash.assert_no_error(ignore_messages=ignore_error_messages)
../../shiningpanda/jobs/1035602e/virtualenvs/d41d8cd9/lib/python3.6/site-packages/widgetastic/widget/base.py:67: in wrapped
    return method(self, *new_args, **new_kwargs)
../../shiningpanda/jobs/1035602e/virtualenvs/d41d8cd9/lib/python3.6/site-packages/widgetastic/utils.py:682: in wrap
    return method(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = SatFlashMessages('//div[@class="toast-notifications-list-pf"]')
ignore_messages = []

    @retry_stale_element
    def assert_no_error(self, ignore_messages=None):
        if ignore_messages is None:
            ignore_messages = []
        self.logger.info('asserting there are no error messages')
        for message in self.messages:
            message_text = message.text
            if message.type not in {'success', 'info', 'warning'}:
                if message_text in ignore_messages:
                    self.logger.info('ERROR MESSAGE IGNORED %s: %r', message.type, message_text)
                    continue
                self.logger.error('%s: %r', message.type, message_text)
                raise AssertionError(
>                   'assert_no_error: {}: {!r}'.format(message.type, message_text))
E               AssertionError: assert_no_error: error: '404 Not Found'

--- a/tests/foreman/ui/test_activationkey.py
+++ b/tests/foreman/ui/test_activationkey.py
@@ -1314,7 +1314,10 @@ def test_positive_delete_manifest(session):
             == DEFAULT_SUBSCRIPTION_NAME
         )
         # Delete the manifest
-        session.subscription.delete_manifest()
+        # Ignore "404 Not Found" as server will connect to upstream subscription service to verify
+        # the consumer uuid, that will be displayed in flash error messages
+        # Note: this happen only when using clone manifest.
+        session.subscription.delete_manifest(ignore_error_messages='404 Not Found')
         assert not session.subscription.has_manifest
         # Verify subscription is not assigned to activation key anymore
         ak = session.activationkey.read(activation_key.name)

